plexus-compilers/plexus-compiler-aspectj/src/main/java/org/codehaus/plexus/compiler/ajc/AspectJCompiler.java

```diff
@@ -565,75 +572,164 @@ private void checkForAspectJRT( List<String> cp )
     private void setSourceVersion( AjBuildConfig buildConfig, String sourceVersion )
         throws CompilerException
     {
-        if ( "11".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK11;
-        }
-        else if ( "10".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK10;
-        }
-        else if ( "9".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK9;
-        }
-        else if ( "1.9".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK9;
-        }
-        else if ( "1.8".equals( sourceVersion ) )
-        {
-            buildConfig.getOptions().sourceLevel = ClassFileConstants.JDK1_8;
-        }
-        else if ( "1.7".equals( sourceVersion ) )
+        buildConfig.getOptions().sourceLevel = versionStringToMajorMinor( sourceVersion );
+    }
+
+    /**
+     * Set the target version in ajc compiler
+     *
+     * @param buildConfig
+     * @param targetVersion
+     */
+    private void setTargetVersion( AjBuildConfig buildConfig, String targetVersion )
+        throws CompilerException
+    {
+        buildConfig.getOptions().targetJDK = versionStringToMajorMinor( targetVersion );
+    }
+
+    private static final Map<String, Long> VERSION_MAP = new HashMap<>();
+
+    static
+    {
+        // Default in Maven Compiler 3.8.1
+        VERSION_MAP.put( "",     JDK1_6 );
+
+        VERSION_MAP.put( "1.1",  JDK1_1 );
+        VERSION_MAP.put( "1.2",  JDK1_2 );
+        VERSION_MAP.put( "1.3",  JDK1_3 );
+        VERSION_MAP.put( "1.4",  JDK1_4 );
```

Comment:
Did aspectj ever support these versions? If not, it is probably safe to remove them.

Reply:
Yes, it does because ACJ does. Remember, AspectJ is also a drop-in replacement for simple Java files without any aspects. This even is what happened before, because the target parameter was not passed on: AJC always created Java 1.2 class files, just like I said in the first sentence of this PR.